subprojects/core/src/main/java/org/gradle/api/internal/DefaultDomainObjectCollection.java

```diff
@@ -391,6 +391,11 @@ public int estimatedSize() {
         return store.estimatedSize();
     }
 
+    @Override
+    public MutationGuard getMutationGuard() {
+        return MutationGuards.of(store);
```

Comment:
why not just `store.getMutationGuard()`? Every store should have a mutation guard, so we don't really need a static method to do some casting for us.